### PR TITLE
rd/ebs_volume - support multi attach

### LINKED ISSUE #13104: [Feature Request] Add multi_attach option to EBS creation.
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

EBS provisioned iops volumes can now be attached to multiple EC2 instances, the multi attach can only be enabled at creation. Adding the option to enable the multi attach would be very useful.

### New or Affected Resource(s)

* aws_ebs_volume

### Potential Terraform Configuration

```hcl
resource "aws_ebs_volume" "ebs_volume" {
    size = 10
    type = "io1"
    iops = 500
    multi_attach = "true"
}
```

### References

https://aws.amazon.com/about-aws/whats-new/2020/02/ebs-multi-attach-available-provisioned-iops-ssd-volumes/

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #13104

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource_aws_ebs_volume: support multi attach
data_source_aws_ebs_volume: support multi attach
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSEBSVolume_'
--- PASS: TestAccAWSEBSVolume_basic (66.31s)
--- PASS: TestAccAWSEBSVolume_updateAttachedEbsVolume (295.79s)
--- PASS: TestAccAWSEBSVolume_updateSize (132.89s)
--- PASS: TestAccAWSEBSVolume_updateType (129.24s)
--- PASS: TestAccAWSEBSVolume_updateIops (135.13s)
--- PASS: TestAccAWSEBSVolume_kmsKey (115.12s)
--- PASS: TestAccAWSEBSVolume_NoIops (70.87s)
--- PASS: TestAccAWSEBSVolume_withTags (63.95s)
--- PASS: TestAccAWSEBSVolume_multiAttach (66.32s)

--- PASS: TestAccAWSEbsVolumeDataSource_basic (77.00s)
--- PASS: TestAccAWSEbsVolumeDataSource_multipleFilters (81.83s)

```

### CODE REVIEW

**Comment by ewbankkit** [comment]
Also add to the equivalent data source?

Verified acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSEBSVolume_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSEBSVolume_ -timeout 120m
=== RUN   TestAccAWSEBSVolume_basic
=== PAUSE TestAccAWSEBSVolume_basic
=== RUN   TestAccAWSEBSVolume_updateAttachedEbsVolume
=== PAUSE TestAccAWSEBSVolume_updateAttachedEbsVolume
=== RUN   TestAccAWSEBSVolume_updateSize
=== PAUSE TestAccAWSEBSVolume_updateSize
=== RUN   TestAccAWSEBSVolume_updateType
=== PAUSE TestAccAWSEBSVolume_updateType
=== RUN   TestAccAWSEBSVolume_updateIops
=== PAUSE TestAccAWSEBSVolume_updateIops
=== RUN   TestAccAWSEBSVolume_kmsKey
=== PAUSE TestAccAWSEBSVolume_kmsKey
=== RUN   TestAccAWSEBSVolume_NoIops
=== PAUSE TestAccAWSEBSVolume_NoIops
=== RUN   TestAccAWSEBSVolume_withTags
=== PAUSE TestAccAWSEBSVolume_withTags
=== RUN   TestAccAWSEBSVolume_multiAttach
=== PAUSE TestAccAWSEBSVolume_multiAttach
=== RUN   TestAccAWSEBSVolume_outpost
--- SKIP: TestAccAWSEBSVolume_outpost (0.00s)
    resource_aws_ebs_volume_test.go:350: Environment variable AWS_OUTPOST_ARN is not set. This environment variable must be set to the ARN of a deployed Outpost to enable this test.
=== CONT  TestAccAWSEBSVolume_basic
=== CONT  TestAccAWSEBSVolume_kmsKey
=== CONT  TestAccAWSEBSVolume_updateIops
=== CONT  TestAccAWSEBSVolume_multiAttach
=== CONT  TestAccAWSEBSVolume_updateType
=== CONT  TestAccAWSEBSVolume_withTags
=== CONT  TestAccAWSEBSVolume_updateSize
=== CONT  TestAccAWSEBSVolume_NoIops
=== CONT  TestAccAWSEBSVolume_updateAttachedEbsVolume
--- PASS: TestAccAWSEBSVolume_NoIops (39.48s)
--- PASS: TestAccAWSEBSVolume_basic (39.80s)
--- PASS: TestAccAWSEBSVolume_multiAttach (39.86s)
--- PASS: TestAccAWSEBSVolume_withTags (40.36s)
--- PASS: TestAccAWSEBSVolume_kmsKey (46.01s)
--- PASS: TestAccAWSEBSVolume_updateIops (69.62s)
--- PASS: TestAccAWSEBSVolume_updateType (69.84s)
--- PASS: TestAccAWSEBSVolume_updateSize (69.85s)
--- PASS: TestAccAWSEBSVolume_updateAttachedEbsVolume (206.27s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	206.315s
```

**Comment by DrFaust92** [comment]
changes to WIP, adding data source

**Comment by DrFaust92** [comment]
added data source support

File: aws/resource_aws_ebs_volume.go [hunk 1]
```diff
@@ -56,6 +56,12 @@ func resourceAwsEbsVolume() *schema.Resource {
 				ForceNew:     true,
 				ValidateFunc: validateArn,
 			},
+			"multi_attach": {
```
[the remark this answers is not in the document]
Comment (DrFaust92) [reply aws/resource_aws_ebs_volume.go:59]: seemed verbose, will fix

File: aws/resource_aws_ebs_volume.go [hunk 2]
```diff
@@ -56,6 +56,12 @@ func resourceAwsEbsVolume() *schema.Resource {
...
+				Type:     schema.TypeBool,
+				Optional: true,
+				Computed: true,
```
[the remark this answers is not in the document]
Comment (DrFaust92) [reply aws/resource_aws_ebs_volume.go:62]: let me take a look

[the remark this answers is not in the document]
Comment (DrFaust92) [reply aws/resource_aws_ebs_volume.go]: removed

File: aws/resource_aws_ebs_volume.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (DrFaust92) [reply aws/resource_aws_ebs_volume.go]: fixed

**Comment by DrFaust92** [comment]
@bflad changes made and tested

**Comment by anGie44** [comment]
@DrFaust92 do you mind adding the attribute re-name change to the data source and website docs as well?

**Comment by DrFaust92** [comment]
🤦  on it.

**Comment by DrFaust92** [comment]
added also disappears test 
```
--- PASS: TestAccAWSEBSVolume_disappears (80.95s)
```

**Pull request merged**